xgcm/test/test_metrics.py

```diff
@@ -283,23 +111,22 @@ def test_assign_metric(key, metric_vars):
         (["X", "Y", "Z"], "tracer", None, ["volume_t"], None),
         (["X"], "u", None, ["dx_e"], None),
         (["X", "Y"], "u", None, ["area_e"], None),
-        (("X", "Y", "Z"), "u", None, ["volume_t"], KeyError),  # This should error out
+        (("X", "Y", "Z"), "u", ["dz_t"], ["volume_t"], KeyError),  # This should error out
```

Comment:
This is the test that I modified. By dropping `dz_t` xgcm can no longer create the volume element at the zonal velocity location and so errors out as expected.

Reply:
This is a bit troubling...why does xgcm reconstruct the volume on the u point from the tracer depth? I think this is a bug.

Reply:
Here xgcm doesn't recreate `volume_t`, it creates the product of `area_e` and `dz_t`. What was happening was the test was failing because the expected `KeyError` was not being raised, so I'm dropping `dz_t` to get the expected error as before.

Reply:
I reverted this one back to how it was before so resolving it...